dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py

```diff
@@ -100,3 +108,21 @@ def raw_script(self) -> str:
             raise PyDSParamException(
                 "Parameter definition do not support % for now.", type(self.definition)
             )
+
+    @property
+    def resource_list(self) -> List:
+        """Get python task define attribute `resource_list`."""
+        if super().resource_list is not None and len(super().resource_list) > 0:
+            for resource in super().resource_list:
+                if (
+                    resource.get("id") is None
+                    and resource.get("resourceName") is not None
+                ):
+                    if resource.get("resourceType") is not None:
+                        resource_type = resource.get("resourceType")
+                    else:
+                        resource_type = "FILE"
+                    resource["id"] = self.query_resource(
+                        resource_type, resource.get("resourceName")
+                    ).get("id")
+        return super().resource_list
```

Comment:
I think we should better put this to `pydolphinscheduler.core.task` instead of python.py. because I know we also support  it in shell task. and in the `core.task` we can add some code like 

```py
    @property
    def resource_list(self) -> List:
        """Get attribute resource_list."""
        resources = set()
        for resource in self._resource_list:
               resources.add(query_resource(resource_type, resource))
        return [{"id": r.get("id")} for r in resources]

    @resource_list.setter
    def resource_list(self, val) -> None:
        """Get attribute resource_list."""
        return self._resource_list = val
```

Reply:
BTW, why we need resource type to query the resources?

Reply:
And if we finish https://github.com/apache/dolphinscheduler/pull/10515#discussion_r903492117, user can not pass `[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}]` to `resource_list` anymore, they can simply pass `["/abce/aa.py", ""/abce/bb.py""]` to it, and we will autofill the `resourceName` keyword adn return with `id`. WDTY?